### ISSUE TITLE
ignoreValueChange prevents suggestions when selecting with keyboard

### ISSUE THREAD

**Issue by tshatrov** [devbridge/jQuery-Autocomplete#755]
I need to implement the following workflow: user types into an input field, selects one of suggestions, then types some more, selects one of suggestions and so on. The use case is an input method for Japanese text, the user types in latin letters and the suggestions are Japanese characters. The delimiter is a regex of non-latin letters, so once a selection is chosen, the user continues to type and the suggestions are queried only for the newly typed text.

This plugin basically works perfectly, except for one little thing. I type something, get suggestions, press DOWN, press ENTER. This causes `ignoreValueChange` to become true. I continue to type something, but I'm not getting suggestions anymore until I trigger `onValueChange` again somehow (I'm using `deferRequestBy` so `onValueChange` triggers only once I finished typing.

Is there a way to circumvent ignoreValueChange being set to true via options or something? My current options are

```
    $('#ichiran-text-field').autocomplete({
        serviceUrl: '/cl/api/ime/',
        paramName: 'w',
        deferRequestBy: 300,
        preventBadQueries: false,
        delimiter: /[^a-zA-Z'\-]+/,
        groupBy: "type",
        triggerSelectOnValidInput: false,
        onSelect: function(suggestion) {this.focus();}
    })
```

**Comment by tkirda** [comment]
Try setting `preserveInput` option to `true`. See if that works for you.

**Comment by tshatrov** [comment]
I tried that, it basically keeps the user input unchanged, which is not what I wanted. In the end I just commented out this line, and I had to add other customisation (2d navigation with up/down moving between groups and left/right moving to next/previous) which I'm not sure if anyone else needs, but the source code is available here https://ichi.moe/js/jquery.autocomplete.js

**Comment by tkirda** [comment]
If `onSelect` is triggered when using keyboard, then probably would make sense to reset `ignoreValueChange` to `false` inside the `onSelect`.

**Comment by tkirda** [comment]
Thanks for the report, and apologies for the very long silence.

The project was dormant for several years and has just shipped **v2.0**, a ground-up TypeScript rewrite with a new build (esbuild), test runner (Vitest + jsdom), and CI pipeline. I'm closing this issue as part of a triage pass on the long-stale backlog — not because the underlying concern lacks merit, but because the surrounding code has changed enough that the original report may no longer apply or may need a fresh repro.

If this is still relevant against current `master`, please open a new issue with a minimal reproduction. I'll review fresh reports against the 2.0 codebase promptly.

Thanks again for taking the time.
